components/locid/src/subtags/region.rs

```diff
@@ -61,6 +62,37 @@ impl Region {
         }
     }
 
+    /// Safely creates a [`Region`] from a reference to its raw format
+    /// as returned by [`Region::into_raw()`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::locid::subtags::Region;
+    ///
+    /// assert!(matches!(Region::try_from_raw(b"US\0"), Ok(_)));
+    /// assert!(matches!(Region::try_from_raw(b"419"), Ok(_)));
+    /// assert!(matches!(Region::try_from_raw(b"foo"), Err(_)));
+    ///
+    /// // Unlike the other constructors, this one is case-sensitive:
+    /// assert!(matches!(Region::try_from_raw(b"us\0"), Err(_)));
+    /// ```
+    pub fn try_from_raw(v: &[u8; 3]) -> Result<&Self, ParserError> {
```

Comment:
question: This is a small array, is there a reason to be dealing in references? My understanding is that it's basically never worth it until it's much, much larger.

Reply:
(comment also posted in https://github.com/unicode-org/icu4x/pull/1694#discussion_r826587004)

Reply:
I remember one reason why I did it on references. I specifically want to assert that the bytes are valid *without changing them in any way*. Existing constructors like `from_bytes` are lenient and mutate the bytes. With borrows, we can guarantee via type safety that the "no change" invariant holds. Returning by value means that we rely on documentation and tests to enforce that invariant.